akka-projection-testkit/src/main/scala/akka/projection/testkit/TestProjection.scala

```diff
@@ -225,19 +322,52 @@ class TestSourceProvider[Offset, Envelope] private (
       else sourceEvents.concat(Source.maybe)
     }
 
+  override def source(offset: Supplier[CompletionStage[Optional[Offset]]])
+      : CompletionStage[akka.stream.javadsl.Source[Envelope, NotUsed]] = {
+    implicit val ec = akka.dispatch.ExecutionContexts.parasitic
+    source(() => offset.get().toScala.map(_.asScala)).map(_.asJava).toJava
+  }
+
   override def extractOffset(envelope: Envelope): Offset = extractOffsetFn(envelope)
 
   override def extractCreationTime(envelope: Envelope): Long = extractCreationTimeFn(envelope)
 }
 
 @ApiMayChange
-class TestInMemoryOffsetStore[Offset](implicit val system: ActorSystem[_]) {
+object TestInMemoryOffsetStore {
```

Comment:
separate TestInMemoryOffsetStore.scala file

Reply:
Deferring to #381 to avoid conflicts.  I've added it to the TODO.